common:
    lid_topic:  "/livox/lidar"
    imu_topic:  "/livox/imu"
    img_topic:  "/cam0/compressed"
    time_sync_en: false         # ONLY turn on when external time synchronization is really not possible
    time_offset_lidar_to_imu: 0.0 # Time offset between lidar and IMU calibrated by other algorithms, e.g. LI-Init (can be found in README).
                                  # This param will take effect no matter what time_sync_en is. So if the time offset is not known exactly, please set as 0.0

preprocess:
    lidar_type: 1                # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR, 
    scan_line: 4
    blind: 0.5

mapping:
    acc_cov: 0.1
    gyr_cov: 0.1
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    fov_degree:    360
    det_range:     100.0
    extrinsic_est_en:  false      # true: enable the online estimation of IMU-LiDAR extrinsic
    extrinsic_T: [ -0.011, -0.02329, 0.04412 ]
    extrinsic_R: [ 1, 0, 0,
                   0, 1, 0,
                   0, 0, 1]

camera:
    ic_extrinsic: [ 0.00684568, -0.999966, -0.00460303, 0.0213688,
                    -0.00396343, 0.00457596, -0.999982, -0.0153776,
                    0.999969, 0.00686379, -0.00393197, -0.0949029,
                    0.0, 0.0, 0.0, 1.0]
    k1: -0.337842 
    k2: 0.111250
    p1: 0.000128
    p2: -0.000393
    fx: 732.068
    fy: 730.743
    cx: 607.391
    cy: 346.292 

    visualize_uncolored: 1
    colored_leaf_size: 0.01
    uncolored_leaf_size: 0.05
    maximum_distance: 2

ceiling_do: true # true: 천장 제거


publish:
    path_en:  false
    scan_publish_en:  true       # false: close all the point cloud output
    dense_publish_en: true       # false: low down the points number in a global-frame point clouds scan.
    scan_bodyframe_pub_en: true  # true: output the point cloud scans in IMU-body-frame

pcd_save:
    pcd_save_en: true
    interval: -1                 # how many LiDAR frames saved in each pcd file; 
                                 # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.
